swaylock/main.c

```diff
@@ -156,7 +157,30 @@ static const struct wl_registry_listener registry_listener = {
 
 static struct swaylock_state state;
 
+bool create_file_lock() {
+	// Env var typically set by logind, e.g. "/run/user/<user-id>"
+	const char *dir = getenv("XDG_RUNTIME_DIR");
+	if (!dir) {
+		dir = "/tmp";
+	}
+    char path[256];
+	if (255 <= snprintf(path, 255,
+			"%s/swaylock.%i.pid", dir, getuid())) {
+		sway_abort("Lock path to long");
+	}
+    int pid_file = open(path, O_CREAT | O_RDWR, 0666);
+    int rc = flock(pid_file, LOCK_EX | LOCK_NB);
+    if(!rc) {
+        return true;
+    }
+    return false;
+}
+
 int main(int argc, char **argv) {
+    if (!create_file_lock()) {
```

Comment:
I think we want to be able to start up multiple sway instances at the same time.

Reply:
Well, it makes no sense to run mutilpe lockscreens in paralell, and it will fail anyway... so I thought it would be nice to handle this properly since this will be triggered if user starts swaylock and then swayidle starts another.